### ISSUE TITLE
Ported twisted.names to Python 3

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8550]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)| @rodrigc reported|
|-|-|
|Trac ID|trac#8550|
|Type|enhancement|
|Created|2016-07-01 20:18:04Z|
|Branch|https://github.com/twisted/twisted/tree/8550-rodrigc-names|

twisted.names needs to be ported to Python 3

<details><summary>Searchable metadata</summary>

```
trac-id__8550 8550
type__enhancement enhancement
reporter__rodrigc rodrigc
priority__normal normal
milestone__Python_3_x Python-3.x
branch__8550_rodrigc_names 8550-rodrigc-names
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__None None
time__1467404284955063 1467404284955063
changetime__1467798034568445 1467798034568445
version__None None
owner__rodrigc rodrigc

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="Lukasa's avatar" src="https://avatars.githubusercontent.com/u/1382556?s=50" width="50" height="50">](https://github.com/Lukasa)<a name="note_3"></a>|@Lukasa set owner to @rodrigc|
|-|-|

The relevant pull request is this one: https://github.com/twisted/twisted/pull/285

This patch is mostly fine. However, I have some concerns about str handling here. Specifically, some interfaces still state that they take `str` where these tests assert that they take `bytes`, which hasn't been resolved. I'd like it if you took a look at those, rodrigc, and either rationalised the string handling or made a convincing argument as to why we shouldn't. =)

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_4"></a>|@rodrigc set owner to @Lukasa|
|-|-|

Review feedback provided:
https://github.com/twisted/twisted/pull/285#discussion_r69658462

**Comment by twisted-trac** [comment]
|[<img alt="Lukasa's avatar" src="https://avatars.githubusercontent.com/u/1382556?s=50" width="50" height="50">](https://github.com/Lukasa)<a name="note_5"></a>|@Lukasa set owner to @rodrigc|
|-|-|

Alright, looks good to me, merge when ready.

**Comment by twisted-trac** [comment]
|<img alt="GitHub's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_6"></a>|GitHub set status to `closed`|
|-|-|

In changeset 51d359c0f86aab35dd309386273084e4bb45e121
```
#!CommitTicketReference repository="" revision="51d359c0f86aab35dd309386273084e4bb45e121"
Merge pull request #285 from twisted/8550-rodrigc-names

Author: rodrigc
Reviewer: Lukasa, glyph
Fixes: #8550

Port twisted.names.resolve and twisted.names.tap to Python 3
```
